lib/opam_solve.ml

```diff
@@ -169,13 +169,19 @@ module Opam_monorepo_context (Base_context : BASE_CONTEXT) :
                    (version, Ok opam_file))
 
   let demote_candidates_to_avoid versions =
-    let avoid_versions, regular_versions =
-      List.partition versions ~f:(fun (_version, opam_res) ->
+    let avoid_versions, broken_versions, regular_versions =
+      List.fold_left versions ~init:([], [], [])
+        ~f:(fun (avoid, broken, regular) ((_, opam_res) as e) ->
```

Comment:
Could you reorder this to be `regular, avoid, broken` to match the final way they get appended?